custom_train.py

```diff
@@ -0,0 +1,212 @@
+        self._TPL_OPTIMIZER = config.get('TRIPLET_TRAINING', 'OPTIMIZER')
+        self._TPL_LR_VALUE = config.getfloat('TRIPLET_TRAINING', 'LR_VALUE')
+        self._TPL_FREEZE_VARS = [
+            var.strip() for var in
+            config.get('TRIPLET_TRAINING', 'FREEZE_VARS').split(',')
+        ]
+
+        # Throw exception if optimizer is not defined
+        if self._CLS_OPTIMIZER not in self._OPTIMIZER_MAP.keys():
+            raise ValueError("CLASSIFIER_TRAINING OPTIMIZER not defined.")
+        if self._TPL_OPTIMIZER not in self._OPTIMIZER_MAP.keys():
+            raise ValueError("TRIPLET_TRAINING OPTIMIZER not defined.")
+
+    def train_classifier(self):
+        '''Train classifer according to specs in config file.'''
+        # When training classifier, we uses one-hot encoding as label
+        self.datset_builder.ONE_HOT = True
+
+        # Create full model using model_builder
+        model, input_name = self.model_builder.create_full_model()
+        # Sanity check
+        model.summary()
+
+        # Set learning rate schedule
+        boundaries = self._CLS_LR_BOUNDARIES
+        values = self._CLS_LR_VALUES
+        lr_schedule = self._LR_SCHEDULE_MAP['PiecewiseConstantDecay'](
+            boundaries=boundaries, values=values)
+        # Use learning reate schedule to create optimizer
+        optimizer = self._OPTIMIZER_MAP[self._CLS_OPTIMIZER](
+            learning_rate=lr_schedule)
+        # Create loss function
+        loss = self._LOSS_MAP['CrossEntropy'](from_logits=True)
+        # Add accuracy metrics
+        accuracy = self._METRIC_MAP['Accuracy']()
+        model.compile(optimizer=optimizer, loss=loss, metrics=[accuracy])
+
+        # Build tf.estimator
+        estimator = tf.keras.estimator \
+            .model_to_estimator(keras_model=model, model_dir=self._CLS_CKPT_DIR)
+        train_spec = tf.estimator.TrainSpec(
+            input_fn=self.datset_builder.get_train_input_fn(input_name),
+            max_steps=self._CLS_MAX_STEP)
+        eval_spec = tf.estimator.EvalSpec(
+            input_fn=self.datset_builder.get_eval_input_fn(input_name))
+
+        # Start training
+        tf.estimator.train_and_evaluate(estimator, train_spec, eval_spec)
+
+    def train_triplet_transfer(self):
+        """Train encoder with triplet loss according to specs in config file."""
+        # When training using triplet loss, we avoid using one-hot encoding
+        self.datset_builder.ONE_HOT = False
+
+        # Create full model using model_builder
+        model, input_name = self.model_builder.create_full_model()
+        # Sanity check
+        model.summary()
+
+        # Build optimizer
+        optimizer = self._OPTIMIZER_MAP[self._TPL_OPTIMIZER](self._TPL_LR_VALUE)
+
+        # Load initial weights from self._TPL_INIT_DIR
+        init_dir = self._TPL_INIT_DIR
+        latest = tf.train.latest_checkpoint(init_dir)
+        model.load_weights(latest)
+
+        # Freeze specified variables
```

Comment:
define a private function freeze_variables and put line 158-166 into that private function and call from here.